main.rb

```diff
@@ -1,47 +1,19 @@
 require_relative 'app'
-
-def show_options
-  puts 'Please choose an option by entering a number:'
-  puts '1 - List all books'
-  puts '2 - List all people'
-  puts '3 - Create a person'
-  puts '4 - Create a book'
-  puts '5 - Create a rental'
-  puts '6 - List all rentals for a given person id'
-  puts '7 - Exit'
-end
+require_relative 'option_handler'
+require_relative 'option'
 
 def prompt
   puts 'Welcome to the School Library App!'
+  option_handler = OptionHandler.new
+  options = ShowOptions.new
+
   loop do
-    show_options
+    options.show_options
     option = gets.chomp.to_i
     break if option == 7
 
-    call_option(option)
-  end
-end
-
-def call_option(option)
-  case option
-  when 1
-    list_books
-  when 2
-    list_people
-  when 3
-    create_person
-  when 4
-    create_book
-  when 5
-    create_rental
-  when 6
-    list_rentals
+    option_handler.call_option(option)
   end
 end
 
-def main
-  app = App.new
-  app.run
-end
-
 main
```

Comment:
- [ ] Please always make sure that your application is working without any error before submitting it for review because as I tried to run your project I got an error related to a function that you are calling at the end of the file. You are trying to call a method which doesn't exist anymore. I think you should use `promp` method instead of `main` to run your application.